src/commands/aem/rde/install.js

```diff
@@ -107,17 +109,78 @@ async function computeStats(url) {
   }
 }
 
+/**
+ *
+ * @param sourceDir
+ * @param outputFilePath
+ */
+async function archiveDirectory(sourceDir, outputFilePath) {
+  const output = fs.createWriteStream(outputFilePath);
+  const archiver = Archiver('zip', { zlib: { level: 9 } });
+
+  archiver.pipe(output);
+
+  await addDirectoryToArchive(archiver, sourceDir, '');
+
+  await archiver.finalize()
+    .then(() => { cli.log(`Finished archiving ${outputFilePath}`); })
+    .catch((err) => { throw err; });
+}
+/**
+
+/**
+ *
+ * @param archiver
+ * @param sourceDir
+ * @param archiveDir
+ */
+async function addDirectoryToArchive(archiver, sourceDir, archiveDir) {
+  const files = fs.readdirSync(sourceDir);
+
+  for (const file of files) {
+    const filePath = path.join(sourceDir, file);
+    const archivePath = path.join(archiveDir, file);
+
+    const stat = fs.lstatSync(filePath);
+    if (stat.isDirectory()) {
+      archiver.file(filePath, { name: archivePath });
+      addDirectoryToArchive(archiver, filePath, archivePath);
+    } else {
+      if (stat.isSymbolicLink()) {
+        const targetPath = fs.readlinkSync(filePath);
+        archiver.symlink(archivePath, targetPath, 0o644);
+      } else {
+        archiver.file(filePath, { name: archivePath });
+      }
+    }
+  }
+}
+
 class DeployCommand extends BaseCommand {
   async run() {
     const { args, flags } = await this.parse(DeployCommand);
     const progressBar = createProgressBar();
 
-    const originalUrl = args.location;
+    let originalUrl = args.location;
+
+    let dispatcherConfigArchiveName = `DISPATCHER-CONFIG-${new Date().toJSON().slice(0, 10)}.zip`;
```

Comment:
Could we move this into the "if" block where it is used?

Reply:
added into the if block